### PR TITLE
SAK-42523 GoogleDrive file picker: Improve picker UI using jsTree

### PR DESCRIPTION
**Pull request by ajordanedf**

### CODE REVIEW

File: content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/AjaxContentEntityProvider.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        return ENTITY_PREFIX;
+    }
+
+    @Override
+    public String[] getHandledOutputFormats() {
+        return new String[] { Formats.JSON };
+    }
+
+    @EntityCustomAction(action = "doNavigateGoogleDrive", viewKey = EntityView.VIEW_LIST)
+    public JSONArray handleChatData(EntityReference view, Map<String,Object> params) {
```
Comment (mpellicer) [inline content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/AjaxContentEntityProvider.java]: I don't think this is a good name for that method.

File: content/content-tool/tool/src/webapp/js/jstree/dist/jstreeButtons.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+					_action = self._create_action("all", actions[i].id);
+					self._set_action(node_id, el, _action);
+				}
+			}
+			return el;
+		};
+
+	}
+
+}));
```
Comment (mpellicer) [inline content/content-tool/tool/src/webapp/js/jstree/dist/jstreeButtons.js]: Please insert an end of file here.

File: entitybroker/tool/src/webapp/WEB-INF/web.xml [hunk 3]
```diff
@@ -6,6 +6,7 @@
   <display-name>sakai-entitybroker-direct</display-name>
   <description>Sakai Direct Entity Access</description>
 
+
```
Comment (mpellicer) [inline entitybroker/tool/src/webapp/WEB-INF/web.xml]: Revert this change.

File: content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm [hunk 4]
```diff
@@ -14,9 +14,14 @@
 		margin: 15px;
 	}
 </style>
+<script> ## This needs to be closed in order for the loads to happen before we use "$"
+includeLatestJQuery('sakai_filepicker_attach.vm');
+</script>
+<script>includeWebjarLibrary("jstree");</script>
```
Comment (mpellicer) [inline content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm]: inside includeWebJarLibrary, pull all the needed imports, .js and .css.

File: library/pom.xml [hunk 5]
```diff
@@ -45,9 +45,15 @@
 			<artifactId>urlrewritefilter</artifactId>
 			<version>4.0.3</version>
 		</dependency>
+		
 		<!-- At the moment we're unpacking the webjars, so we don't want them bundled into the WAR
 		     but we still need them as dependencies so the plugin unpacks them, that's why they
 		     have a scope of runtime -->
+		<dependency>
```
Comment (mpellicer) [inline library/pom.xml]: Move the dependency above the comment, the comment belongs to the next dependency.

File: content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/AjaxContentEntityProvider.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.content.api.ContentTypeImageService;
+import static org.sakaiproject.content.tool.FilePickerAction.STATE_GOOGLEDRIVE_CHILDREN;
```
Comment (mpellicer) [inline content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/AjaxContentEntityProvider.java]: Are these dependencies needed?

File: content/content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java [hunk 7]
```diff
@@ -325,7 +326,6 @@ else if(MODE_ATTACHMENT_EDIT_ITEM.equals(helper_mode))
 		{
 			helper_mode = MODE_ATTACHMENT_EDIT_ITEM_INIT;
 		}
-
```
Comment (mpellicer) [inline content/content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java]: Revert this change

File: content/content-tool/tool/src/webapp/WEB-INF/applicationContext.xml [hunk 8]
```diff
@@ -24,5 +24,14 @@
         <property name="userDirectoryService" ref="org.sakaiproject.user.api.UserDirectoryService"/>
         <property name="entityManager" ref="org.sakaiproject.entity.api.EntityManager" />
     </bean>
+
```
Comment (mpellicer) [inline content/content-tool/tool/src/webapp/WEB-INF/applicationContext.xml]: Remove extra space and fix indenting.

File: content/content-tool/tool/src/webapp/js/jstree/dist/jstreeButtons.js [hunk 9]
```diff
@@ -0,0 +1,194 @@
+(function (factory) {
```
Comment (mpellicer) [inline content/content-tool/tool/src/webapp/js/jstree/dist/jstreeButtons.js]: Can you document why this file is needed? Also try to use standard indenting.

File: content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm [hunk 10]
```diff
@@ -14,9 +14,14 @@
...
+<script>includeWebjarLibrary("jstree/3.2.1/themes/default/style.min.css");</script>
+<script src="/sakai-content-tool/js/jstree/dist/jstreeButtons.js"></script>
 
 <script type="text/javascript">
-	<!--
```
Comment (mpellicer) [inline content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm]: Why do you uncomment this?

File: content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm [hunk 11]
```diff
@@ -34,7 +39,75 @@
 				theForm.submit();
 			}
 		}
-	//-->
+
+		jQuery(function() {
```
Comment (mpellicer) [inline content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm]: Better indenting required.

File: content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/AjaxContentEntityProvider.java (hunk 1 above)
Comment (ajordanedf) [reply content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/AjaxContentEntityProvider.java]: I have changed the name, how do you think about the new?
